### PR TITLE
Iac 733 add support for overwrite vrli content packs (Duplicate to https://github.com/vmware/build-tools-for-vmware-aria/pull/32) needs to be declined

### PR DESCRIPTION
**Pull request by akantchev**
### Checklist

- [x] Lint and unit tests pass locally with my changes
- [ ] I have added relevant error handling and logging messages to help troubleshooting
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have updated CHANGELOG.md with a short summary of the changes introduced
- [x] I have tested against live environment, if applicable
- [ ] I have added relevant usage information (As-built)
- [ ] Any structure and/or content vRA-NG improvements are synchronized with vra-ng and ts-vra-ng archetypes
- [x] Every new or updated Installer property is documented in docs/archive/doc/markdown/use-bundle-installer.md
- [x] Dependencies in pom.xml are up-to-date
- [x] My changes have been rebased and squashed to the minimal number of relevant commits
- [x] My changes have a descriptive commit message with a short title, including a `Fixed #XXX -` or `Closed #XXX -` prefix to auto-close the issue

### CODE REVIEW

File: docs/archive/doc/markdown/use-bundle-installer.md [hunk 1]
```diff
@@ -121,6 +121,7 @@ vrli_vrops_server_port
 vrli_vrops_server_user
 vrli_vrops_server_password
 vrli_vrops_server_auth_source
+vrli_package_import_overwrite_mode
```
Comment (Michaelpalacce) [inline docs/archive/doc/markdown/use-bundle-installer.md:124]: Don't modify the old documentation. Lets add this to the new one?

File: common/artifact-manager/src/main/java/com/vmware/pscoe/iac/artifact/rest/client/vrli/AbstractRestClientVrli.java [hunk 2]
```diff
@@ -33,114 +33,120 @@
 import java.util.Properties;
 
 public abstract class AbstractRestClientVrli extends RestClient {
```
Comment (Michaelpalacce) [inline common/artifact-manager/src/main/java/com/vmware/pscoe/iac/artifact/rest/client/vrli/AbstractRestClientVrli.java:35]: Can you revert the tab -> whitespace changes, can't review it like this and it was not necessary

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,7 +1,9 @@
 ## v2.29.2 - 24 Feb 2023
 
 ### Enhancements
+* [maven-plugins-vrli] IAC-733 / Add support for overwriting of already existing vRLI content packs.
```
Comment (Michaelpalacce) [inline CHANGELOG.md:4]: Add this to the top of the changelog ( above the version )

**Comment by Michaelpalacce** [comment]
Can you revert all tab->whitespaces changes

File: common/artifact-manager/src/main/java/com/vmware/pscoe/iac/artifact/rest/client/vrli/RestClientVrliV2.java [hunk 4]
```diff
@@ -38,29 +37,31 @@
 import java.util.stream.Collectors;
 
 public class RestClientVrliV2 extends AbstractRestClientVrli {
```
Comment (Michaelpalacce) [inline common/artifact-manager/src/main/java/com/vmware/pscoe/iac/artifact/rest/client/vrli/RestClientVrliV2.java:39]: I think it's prudent for V2 to extend V1 and do overwrites where needed?

File: common/artifact-manager/src/main/java/com/vmware/pscoe/iac/artifact/rest/client/vrli/AbstractRestClientVrli.java [hunk 5]
```diff
@@ -33,114 +33,120 @@
...
+    protected static final String CONTENT_PACKS_LIST_API = "/content/contentpack/list";
+    protected static final String ALERTS_API = "/alerts";
+    protected static final String VRLI_RESOURCE_KEY_TYPE = "LogInsightLogServer";
+    private static final String OVERWRITE_MODE = "OVERWRITE";
+
+    protected AbstractRestClientVrli(String apiPrefix, ConfigurationVrli configuration, RestTemplate restTemplate) {
+        this.apiPrefix = apiPrefix;
+        this.restTemplate = restTemplate;
+        this.configuration = configuration;
+        this.vropsRestClient = getVropsRestClient();
```
Comment (Michaelpalacce) [inline common/artifact-manager/src/main/java/com/vmware/pscoe/iac/artifact/rest/client/vrli/AbstractRestClientVrli.java:52]: Why is this needed?

File: common/artifact-manager/src/main/java/com/vmware/pscoe/iac/artifact/rest/client/vrli/RestClientVrliV2.java (hunk 4 above)
Comment (konstantin-nachev) [reply common/artifact-manager/src/main/java/com/vmware/pscoe/iac/artifact/rest/client/vrli/RestClientVrliV2.java:39]: The idea is that V2 will be totally different in term of models. From my perspective doesn't make sense.

File: common/artifact-manager/src/main/java/com/vmware/pscoe/iac/artifact/rest/client/vrli/AbstractRestClientVrli.java (hunk 5 above)
Comment (akantchev) [reply common/artifact-manager/src/main/java/com/vmware/pscoe/iac/artifact/rest/client/vrli/AbstractRestClientVrli.java:52]: For rewriting the data for the vCOPs enabled alerts (recipients->vrops->vcopsResourceKindKey)

File: common/artifact-manager/src/main/java/com/vmware/pscoe/iac/artifact/rest/client/vrli/RestClientVrliV2.java (hunk 4 above)
Comment (akantchev) [reply common/artifact-manager/src/main/java/com/vmware/pscoe/iac/artifact/rest/client/vrli/RestClientVrliV2.java:39]: I agree with @konstantin-nachev

File: docs/archive/doc/markdown/use-bundle-installer.md (hunk 1 above)
Comment (akantchev) [reply docs/archive/doc/markdown/use-bundle-installer.md:124]: Updating the installer docs is stated in the common task-list for the PR
